.expeditor/integration_test.pipeline.sh

```diff
@@ -0,0 +1,294 @@
+    [[ -z "$TF_VAR_enable_ipv6" ]] && TF_VAR_enable_ipv6="${ENABLE_IPV6:-true}"
+    [[ -z "$TF_VAR_platform" ]] && TF_VAR_platform="${PLATFORM:-ubuntu-18.04}"
+
+    if [[ "$TF_VAR_enable_ipv6" = 'true' ]]; then
+        workspace="${BUILD_NUMBER}-${TF_VAR_scenario}-ipv6-${TF_VAR_platform}"
+    else 
+        workspace="${BUILD_NUMBER}-${TF_VAR_scenario}-ipv4-${TF_VAR_platform}"
+    fi
+  fi
+
+  export TF_VAR_scenario
+  export TF_VAR_enable_ipv6
+  export TF_VAR_platform
+
+  # ensure the workspace is available
+  setup "$TF_VAR_scenario" "$TF_VAR_enable_ipv6" "${TF_VAR_platform}"
+
+  echo "--- Destroying $workspace"
+
+  # set bogus values for destroy
+  export TF_VAR_install_version_url='NULL'
+  export TF_VAR_upgrade_version_url='NULL'
+  export TF_VAR_backend_version_url='NULL'
+
+  terraform destroy -auto-approve || error 'terraform destroy failed! Manual cleanup of resources may be required'
+  terraform workspace select default || error 'terraform failed to switch to the "default" workspace!'
+  terraform workspace delete "$workspace" || error "terraform failed to delete the $workspace workspace! Manual cleanup of resources may be required"
+}
+
+# destroy all terraform scenarios
+destroy-all () {
+  local workspace
+  local ret=0
+
+  echo "--- Destroying all Terraform workspaces associated with build number $BUILD_NUMBER"
+
+  # verify command dependencies
+  [[ "$(command -v consul)" ]] || error 'consul command is not available'
+
+  # iterate across workspaces left in consul
+  for workspace in $(consul kv get -keys terraform/chef-server/ | sed "s/.*:\(${BUILD_NUMBER}-.*$\)/\1/;s/\///;" | sort -u); do
+    destroy "$workspace" || ret=1
+  done
+
+  exit $ret
+}
+
+echo '--- Verifying environment'
+
+# allow expeditor to override ACTION when a cancellation or failure occurred
+[[ "$EXPEDITOR_BUILD_STATE" =~ ^(canceled|failed)$ ]] && ACTION='destroy'
+
+# allow for environment override of action
+case "$ACTION" in
+  apply)
+    action='apply'
+    ;;
+  destroy)
+    action='destroy'
+    ;;
+  destroy-all)
+    action='destroy-all'
+    ;;
+esac
+
+# verify we have an action set
+[[ -z "$action" ]] && error 'no action provided'
+
+# verify command dependencies
+[[ "$(command -v terraform)" ]] || error 'terraform command is not available'
+
+# verify environment variables
+[[ -z "$BUILDKITE_BUILD_NUMBER" ]] && error 'BUILDKITE_BUILD_NUMBER environment variable is required!'
+[[ -z "$BUILDKITE_LABEL" ]] && error 'BUILDKITE_LABEL environment variable is required!'
+
+# allow for environment override of build number
+[[ -z "$BUILD_NUMBER" ]] && BUILD_NUMBER="${EXPEDITOR_BUILD_NUMBER}"
+[[ -z "$BUILD_NUMBER" ]] && BUILD_NUMBER="${BUILDKITE_BUILD_NUMBER}"
+export BUILD_NUMBER
+
+# apply defaults if necessary
+[[ -z "$TF_VAR_build_prefix" ]] && TF_VAR_build_prefix="${BUILD_NUMBER}-"
+export TF_VAR_build_prefix
+[[ -z "$TF_VAR_aws_contact" ]] && TF_VAR_aws_contact="${AWS_CONTACT:-releng}"
```

Comment:
The default contact shouldn't be releng -- it should be someone on the Chef Server team.

Reply:
@tduffield I used `releng` as the default value as that is what would show up as the contact for the resources created during a buildkite run.  This seemed appropriate since the resources are being created in a VPC in the `chef-cd` account that a chef-server developer wouldn't have access to clean up if something went wrong.  My attempt was to follow a similar pattern to the instances that are normally created by other buildkite runs in the `delivered-default` VPC.
Would it be preferred to use `buildkite` as the contact tag rather than `releng`?  Or should I really just do `csnapp`?

Reply:
So the "owner" are the individuals who can provide context on how/why certain infrastructure is being used. While the infra may run in an account managed by releng, we don't "own" it per say. If anything in that situation we'd just be a middleman.

Reply:
I've updated the contact to `chef-eng-team` as recommended by @OBrienCommaJosh